src/orion/core/io/cmdline_parser.py

```diff
@@ -91,9 +91,26 @@ def format(self, configuration):
         # Arguments with spaces are broken if we use ' '.join(template).format().split(' ').
         # Hence we iterate over the list as-is and format on each item separately.
         formatted = []
+        item: str
         for item in self.template:
             if item.startswith("-"):
                 formatted.append(item)
+            elif (
```

Comment:
Is this part still needed for the flatten() in orion_cmdline_parser solves the issue? The generic parser should not need to dig into the configuration to find the item in other nested dict.

Reply:
Yes, this is still needed:
1. potential `"{trial.hash_id}"`-like interpolations
2. `"{a.b}".format(**{"a.b": 123})` doesn't work, so we need an additional check here.

> The generic parser should not need to dig into the configuration to find the item in other nested dict

I'm not sure I understand the last part, I don't see any digging happening here, would you mind clarifying this a bit?
Was this comment from before the latest updates, when there was still my hacky code in there?

Reply:
It's more clear thanks! The issue is that `format()` tries to assign a variable `a` with attribute `b`. This problem comes from the fact that the key value is defined by the user, and it could occur with other characters than `.`.

We would also need to adapt the command here for the tests of CmdlineParser to verify such cases: https://github.com/Epistimio/orion/blob/develop/tests/unittests/core/io/test_cmdline_parser.py#L53.

I think the best would be to shield CmdlineParser from weird user argument and use an internal dict mapping from user keys to safe keys, so that the templates we build during `CmdlineParser.parse()` only contain safe keys (no invalid characters for `format()`). That would require building the mapping dict during `parse()`, handle backward compatibility in `state_dict`/`set_state_dict` and replace configuration keys during `format()`. Another solution would be to use curated keys, for which we would remove the problematic characters, but then we risk causing collisions if the user had arguments like --ab and --a.b in the same commandline call.

That's a lot more work so maybe it can be delayed after the sprint?

Reply:
Some other characters will lead to the same issue, like `--some:weird arg-name`. Clearly `.` will be the most common source of issue, but the problem we have with it is a symptom of a more general problem.

Reply:
Alright, but this PR is a simple fix for a simple bug, I think it's best if we tackle other issues of this type in another PR, wouldn't you agree?

Reply:
It's an adhoc fix for one symptom of a bigger problem. Another PR would aim at fixing the same issue, but at the root of it.

Reply:
It's not super urgent however, so maybe a temporary plaster solution is better than waiting to solve the main source of the issue. We can keep trace of it in an issue.

Reply:
Well, looks like you fixed it all! :)

Reply:
Fixed in  5f3d3b840f477524acb8fec70b4b1a5db565416b: Made the check more general, now works with a whole variety of weird argument names, not just those with ".".